src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -211,33 +211,19 @@ public sealed override string Name
         }
 
         internal override LexicalSortKey GetLexicalSortKey()
-        {
-            return new LexicalSortKey(_location, this.DeclaringCompilation);
-        }
+            => new LexicalSortKey(_syntaxReference, this.DeclaringCompilation);
+
+        public override Location TryGetFirstLocation()
+            => _syntaxReference.SyntaxTree.GetLocation(_locationSpan);
```

Comment:
Is there any risk that certain access patterns here could actually make us allocate more of these?

Reply:
I thought the location is actually used for local variables in order to report "use before declaration" errors. Maybe it is already working in terms of spans.

It seems very plausible that almost all of the time there is no error reported on a local, and if there is, it is usually 1 error. Thanks!